api/src/test/java/org/pmiops/workbench/db/dao/DataSetServiceTest.java

```diff
@@ -0,0 +1,224 @@
+          .addNamedParameter(
+              "foo",
+              QueryParameterValue.newBuilder()
+                  .setType(StandardSQLTypeName.INT64)
+                  .setValue(Long.toString(101L))
+                  .build())
+          .build();
+
+  @Autowired private BigQueryService bigQueryService;
+  @Autowired private CdrBigQuerySchemaConfigService cdrBigQuerySchemaConfigService;
+  @Autowired private CohortDao cohortDao;
+  @Autowired private ConceptSetDao conceptSetDao;
+  @Autowired private ConceptBigQueryService conceptBigQueryService;
+  @Autowired private CohortQueryBuilder cohortQueryBuilder;
+  @Autowired private DataSetDao dataSetDao;
+
+  private DataSetServiceImpl dataSetServiceImpl;
+
+  @TestConfiguration
+  @MockBean({
+    BigQueryService.class,
+    CdrBigQuerySchemaConfigService.class,
+    CohortDao.class,
+    ConceptBigQueryService.class,
+    ConceptSetDao.class,
+    CohortQueryBuilder.class,
+    DataSetDao.class
+  })
+  static class Configuration {}
+
+  @Before
+  public void setUp() {
+    dataSetServiceImpl =
+        new DataSetServiceImpl(
+            bigQueryService,
+            cdrBigQuerySchemaConfigService,
+            cohortDao,
+            conceptBigQueryService,
+            conceptSetDao,
+            cohortQueryBuilder,
+            dataSetDao);
+
+    final Cohort cohort = buildSimpleCohort();
+    when(cohortDao.findCohortByNameAndWorkspaceId(anyString(), anyLong())).thenReturn(cohort);
+    when(cohortQueryBuilder.buildParticipantIdQuery(any()))
+        .thenReturn(
+            QueryJobConfiguration.newBuilder(
+                    "SELECT * FROM person_id from `${projectId}.${dataSetId}.person` person")
+                .build());
+    when(bigQueryService.filterBigQueryConfig(any(QueryJobConfiguration.class)))
+        .thenReturn(QUERY_JOB_CONFIGURATION_1);
+  }
+
+  private Cohort buildSimpleCohort() {
+    final SearchRequest searchRequest = SearchRequests.males();
+    final String cohortCriteria = new Gson().toJson(searchRequest);
+
+    final Cohort cohortDbModel = new Cohort();
+    cohortDbModel.setCohortId(101L);
+    cohortDbModel.setType("foo");
+    cohortDbModel.setWorkspaceId(1L);
+    cohortDbModel.setCriteria(cohortCriteria);
+    return cohortDbModel;
+  }
+
+  private static DataSetRequest buildEmptyRequest() {
+    final DataSetRequest invalidRequest = new DataSetRequest();
+    invalidRequest.setValues(Collections.emptyList());
+    return invalidRequest;
+  }
+
+  private static DataSetRequest buildTrivialRequest() {
+    final DataSetRequest result = new DataSetRequest();
+    result.setCohortIds(ImmutableList.of(COHORT_ID_1));
+    result.setConceptSetIds(ImmutableList.of(CONCEPT_SET_ID_1));
+    result.setValues(Collections.emptyList());
+    result.setValues(ImmutableList.of(DOMAIN_VALUE_PAIR_PERSON_PERSON_ID));
+    return result;
+  }
+
+  @Test(expected = BadRequestException.class)
+  public void itThrowsForNoCohortOrConcept() throws Exception {
```

Comment:
Consistency nit: I think most if not all of our other test classes use the "testFoo" form, not the "itFoo" form. Unless there's a strong functional reason to phrase things this way, I'd prefer consistency over personal style here.

Reply:
They beat this into me at HubSpot. I think the reason is that it's an assertion in active voice, so if you see the failure in the log you know exactly what the test case is trying to do. It's harder to start a sentence with `test` that's as crisp. Both prefixes are redundant b/c the framework knows which are test methods based on the annotation.

I'll change these though.